### ISSUE TITLE
Mistake promise/prefer-await-to-callback

### ISSUE THREAD

**Issue by mathieug** [xjamundx/eslint-plugin-promise#116]
How do I avoid an error with `promise/prefer-await-to-callback` for this following code which use events:
```javascript
dbConn.on('error', (err) => logger.warn('MongoDB event `error`:', err));
```

**Comment by macklinu** [comment]
For now, I would disable that rule for the line using:

```js
// eslint-disable-next-line promise/prefer-await-to-callbacks
dbConn.on('error', (err) => logger.warn('MongoDB event `error`:', err));
```

This seems like a bug though. Perhaps that rule should ignore callbacks in an `on()` function (which indicates using events). That's not something we currently check for but should probably be considered. I'm not 100% sure how to accomplish that off the top of my head, but suggestions or PRs are welcome!

**Comment by macklinu** [comment]
I think I found a fix for this - pushing up to #117.

**Comment by mathieug** [comment]
Maybe this plugin can help you: [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn), check the rule `prefer-add-event-listener` for example.

**Comment by alexilyaev** [comment]
Also the rule page is missing examples:  
https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/prefer-await-to-callbacks.md
